rune/libenclave/internal/runtime/pal/pal_linux.go

```diff
@@ -16,15 +20,49 @@ const (
 	palApiVersion = 3
 )
 
+var enclavePoolType string
+var isEncCache bool
+var hasEpm bool
+
 func (pal *enclaveRuntimePal) Init(args string, logLevel string) error {
+	var enclaveinfo *v1alpha1.Enclave
+	var err error
+	var addr int64 = 0
+	var fd int = 0
+
+	hasEpm = false
+	isEncCache = false
 	api := &enclaveRuntimePalApiV1{}
 	ver := api.get_version()
 	if ver > palApiVersion {
 		return fmt.Errorf("unsupported pal api version %d", ver)
 	}
 	pal.version = ver
-
-	return api.init(args, logLevel)
+	if strings.Contains(args, "epm") {
+		hasEpm = true
+	}
+	if hasEpm == true {
+		if strings.Contains(args, "skeleton") {
+			enclavePoolType = "enclave-cache.skeleton"
+		} else if strings.Contains(args, "occlum") {
+			enclavePoolType = "enclave-cache.occlum"
+		}
+
+		enclaveinfo = epm.GetEnclave(enclavePoolType)
+		if enclaveinfo != nil {
+			isEncCache = true
+			epm.SgxMmap(*enclaveinfo)
+			args = args + " sgxmmap"
+			addr = enclaveinfo.Layout[0].Addr
```

Comment:
How to guarantee layout[0] always contains the minimum address?

Reply:
layout info is from  /proc/pid/maps. In maps, the addr is sorted from low addr to high addr. So after parsing maps,  it can guarantee layout[0] is the minimum address for /dev/sgx/enlcave mmaping.

Reply:
Sounds good. Plz add the comment lines to describe.